sys/src/9/pc/l.s

```diff
@@ -671,9 +671,19 @@ TEXT lcycles(SB),1,$0
 	RDTSC
 	RET
 
+// WIP: if the rdmsr fails, then return all -1
+// TODO: change type signature of rdmsr
 TEXT rdmsr(SB), $0				/* model-specific register */
 	MOVL	index+0(FP), CX
+TEXT rdmsr_doit(SB), $0
 	RDMSR
+TEXT rdmsr_ok(SB), $0
```

Comment:
I don't think you need this label. But I'd probably do the inverse; assume success, and have the failure case after tha mainline code, and then have it jump back to  the exit:

```
TEXT rdmsr(SB), $0
        MOVL    index+0(FP), CX
TEXT mayberdmsr(SB), $0
        RDMSR
1:
	MOVL	vlong+4(FP), CX			/* &vlong */
	MOVL	AX, 0(CX)			/* lo */
	MOVL	DX, 4(CX)			/* hi */
	RET
TEXT rdmsrfail(SB), $0
	MOVL    $0xffffffff, AX
	MOVL	AX, DX
        JMP     1b
```